sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -384,21 +386,75 @@ class GpuCoalesceIterator(iter: Iterator[ColumnarBatch],
     concatTime,
     totalTime,
     peakDevMemory,
-    opName) {
+    opName) with Arm {
 
   private var batches: ArrayBuffer[ColumnarBatch] = ArrayBuffer.empty
   private var maxDeviceMemory: Long = 0
 
-  override def initNewBatch(): Unit =
-    batches = ArrayBuffer[ColumnarBatch]()
+  // batch indices that are compressed batches
+  private[this] var compressedBatchIndices: ArrayBuffer[Int] = ArrayBuffer.empty
 
-  override def addBatchToConcat(batch: ColumnarBatch): Unit =
+  private[this] var codec: TableCompressionCodec = _
+
+  override def initNewBatch(): Unit = {
+    batches.clear()
+    compressedBatchIndices.clear()
+  }
+
+  override def addBatchToConcat(batch: ColumnarBatch): Unit = {
+    if (isBatchCompressed(batch)) {
+      compressedBatchIndices += batches.size
+    }
     batches += batch
+  }
+
+  private def isBatchCompressed(batch: ColumnarBatch): Boolean = {
+    if (batch.numCols == 0) {
+      false
+    } else {
+      batch.column(0) match {
+        case _: GpuCompressedColumnVector => true
+        case _ => false
+      }
+    }
+  }
 
-  override def getColumnSizes(cb: ColumnarBatch): Array[Long] =
-    GpuColumnVector.extractBases(cb).map(_.getDeviceMemorySize)
+  private def getUncompressedColumnSizes(tableMeta: TableMeta): Array[Long] = {
+    val numCols = tableMeta.columnMetasLength
+    val columnMeta = new ColumnMeta
+    val subBufferMetaObj = new SubBufferMeta
+    val sizes = new Array[Long](numCols)
+    (0 until numCols).foreach { i =>
+      tableMeta.columnMetas(columnMeta, i)
+      var subBuffer = columnMeta.data(subBufferMetaObj)
```

Comment:
This doesn't compile.  `columnMeta.data` has a both parameterless form and a single-parameter form, therefore Scala thinks it is a sequence of `SubBufferMeta` which is not what we want.  I get what you're trying to suggest here, but I'm not sure how to easily get Scala to reference the proper Java method because of the overloads.

Reply:
Thanks for trying, sorry I should have tried to compile this.